rllib/core/rl_trainer/tests/tf/test_tf_rl_trainer.py

```diff
@@ -0,0 +1,179 @@
+import gym
+import unittest
+
+import ray
+
+from ray.rllib.algorithms import AlgorithmConfig
+from ray.rllib.offline import IOContext
+from ray.rllib.offline.dataset_reader import (
+    DatasetReader,
+    get_dataset_and_shards,
+)
+
+from ray.rllib.core.rl_trainer.trainer_runner import TrainerRunner
+from ray.rllib.core.rl_trainer.tf.tf_rl_trainer import TfRLTrainer
+from ray.rllib.core.testing.tf.bc_module import DiscreteBCTFModule
+from ray.rllib.core.testing.tf.bc_optimizer import BCTFOptimizer
+from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID, MultiAgentBatch
+
+
+class TestTfRLTrainer(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        ray.init()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        ray.shutdown()
+
+    def test_update_multigpu(self):
+        """Test training in a 2 gpu setup and that weights are synchronized."""
+        env = gym.make("CartPole-v1")
+        trainer_class = TfRLTrainer
+        trainer_cfg = dict(
+            module_class=DiscreteBCTFModule,
+            module_config={
+                "observation_space": env.observation_space,
+                "action_space": env.action_space,
+                "model_config": {"hidden_dim": 32},
+            },
+            optimizer_class=BCTFOptimizer,
+            optimizer_config={},
+            debug=True,
```

Comment:
remember to remove this flag?

Reply:
I'm purposely doing this because this flag enables trainer and runner to return info that can be used during testing / debugging.

Reply:
ok no problem.